lib/hammer_cli/modules.rb

```diff
@@ -76,14 +76,45 @@ def self.require_module(name)
     end
 
     def self.load_all
-      HammerCLI::Modules.names.each do |m|
+      module_names = HammerCLI::Modules.names
+      module_names.each do |m|
         Modules.load(m)
       end
       loaded_for_deps = loaded_modules & disabled_modules
       unless loaded_for_deps.empty?
         message = _("Error: Some of the required modules are disabled in configuration: %s.") % loaded_for_deps.join(', ')
         raise HammerCLI::ModuleDisabledButRequired.new(message)
       end
+      return if modules_checksum
+
+      File.open(checksum_file_path, 'w') do |f|
+        f.write(compute_modules_checksum(module_names))
+      end
+    end
+
+    def self.module_version(mod)
+      Gem.loaded_specs[mod]&.version&.to_s
+    end
+
+    def self.compute_modules_checksum(names)
+      mods_with_version = names.map do |m|
+        "#{m}-#{module_version(m)}"
+      end.join(',')
+      Digest::SHA1.hexdigest(mods_with_version)
+    end
+
+    def self.modules_checksum
+      return unless File.exist?(checksum_file_path)
+
+      File.read(checksum_file_path)
+    end
+
+    def self.changed?
+      modules_checksum != compute_modules_checksum(HammerCLI::Modules.names)
+    end
+
+    def self.checksum_file_path
+      @checksum_file_path ||= File.expand_path(HammerCLI::Settings.get(:checksum_cache_file))
```

Comment:
Can this setting be unset?

Reply:
Yes, but the supported way is to simply comment the setting out in the config file. In this case defaults will be used, so the value of this setting will be always present.

The only way to break this is to set `:checksum_cache_file: nil` in the config file, but AFAIK this never happened before, otherwise we would have updated the way we treat unsetting cache file paths.